edm4hep.yaml

```diff
@@ -338,9 +338,10 @@ datatypes:
     Author: "EDM4hep authors"
     Members:
       - int32_t   PDG                        // PDG code of the shower particle that caused this contribution
-      - float energy [GeV]                    // energy of this contribution
-      - float time [ns]                     // time of this contribution
-      - edm4hep::Vector3f stepPosition [mm] // position of this energy deposition (step)
+      - float energy [GeV]                   // energy of this contribution
+      - float time [ns]                      // time of this contribution
+      - edm4hep::Vector3f stepPosition [mm]  // position of this energy deposition (step)
+      - float stepLength [mm]                // Geant4 step length for this contribution
     OneToOneRelations:
       - edm4hep::MCParticle particle     // primary MCParticle that caused the shower responsible for this contribution to the hit
```

Comment:
While this is not directly part of this PR, shall we also change this comment "primary MCParticle that caused the shower responsible for this contribution to the hit" to "MCParticle that caused this hit contribution" because it looks like DDSIM does not ensure this (https://github.com/AIDASoft/DD4hep/blob/master/DDG4/edm4hep/Geant4Output2EDM4hep.cpp#L626-L630)? Or we change this behavior in DD4hep?

Reply:
This translates to the "primary" MCParticle, i.e. the particle that is still stored in the MCParticle record.

https://github.com/AIDASoft/DD4hep/blob/c7fc3e9ae29c1641c02c2aaf6b93b4fb321f0504/DDG4/edm4hep/Geant4Output2EDM4hep.cpp#L626-L627

Reply:
"MCParticle that caused the hit. NB: if the MCParticle causing the hit is not kept in the MCParticle record, this will point to the first mother actually present in the output"?

Reply:
Done with slight changes to your proposal. Let me know if you think this works.